### PR TITLE
Danh/add-more-pair-hardcode

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
- Add more token pairs to hardcode list
- Small refactor my code: split code to appropriate file, ...

### CODE REVIEW

File: src/components/swapv2/TokenInfoV2.tsx [hunk 1]
```diff
@@ -26,33 +26,29 @@ const checkTokenDescription = ({
   chainId: ChainId | undefined
 }) => {
   // hard code pair description for SEO
-  const rs1 = JSON.parse(JSON.stringify(tokenInfo1))
-  const rs2 = JSON.parse(JSON.stringify(tokenInfo2))
-  let isHardCode = false
+  const rs1: TokenInfo = JSON.parse(JSON.stringify(tokenInfo1))
```
Comment (hungdoansy) [inline src/components/swapv2/TokenInfoV2.tsx:29]: This is not a good way to deep-clone an object. Link: https://stackoverflow.com/a/5344074/6812545

We should use `immer` here.

File: src/utils/string.ts [hunk 2]
```diff
@@ -34,3 +35,21 @@ export const convertSymbol = (network: string, value: string) => {
...
 }
+
+/**
+ * check url format /network/sym1-to-sym2, sym1 vs sym2 is in whitelist
+ * @param chainId
+ * @param symbol1 ex: knc
+ * @param symbol2 ex: usdt
+ * @returns
+ */
+export const checkPairInWhiteList = (chainId: ChainId | undefined, symbol1: string, symbol2: string) => {
```
Comment (hungdoansy) [inline src/utils/string.ts:46]: This func should be moved to `utils/token.ts` or `utils/pair.ts` instead. string is not related here

File: src/components/swapv2/TokenInfoV2.tsx (hunk 1 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/TokenInfoV2.tsx:29]: ah, after review. I see there is no need to deep copy.  I update it to 
image
thanks for you suggestion.

Comment (hungdoansy) [reply src/components/swapv2/TokenInfoV2.tsx:29]: Yeah, even in that case, you can consider using immer. It'd be much more readable

File: src/pages/SwapV2/index.tsx [hunk 3]
```diff
@@ -529,7 +527,7 @@ export default function Swap({ history }: RouteComponentProps) {
     findTokenPairFromUrl()
     refIsImportUserToken.current = false
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [JSON.stringify(defaultTokens), refIsCheckNetworkAutoSelect.current])
+  }, [defaultTokens, refIsCheckNetworkAutoSelect.current])
```
Comment (hungdoansy) [inline src/pages/SwapV2/index.tsx:530]: Please add a comment here why you need `defaultTokens` here though it's not in use

File: src/pages/SwapV2/index.tsx [hunk 4]
```diff
@@ -564,7 +562,11 @@ export default function Swap({ history }: RouteComponentProps) {
         )}&networkId=${chainId}`
       : window.location.origin + `/swap?networkId=${chainId}`
 
-  const renderTokenInfo = Boolean(isShowTokenInfoSetting && (currencyIn || currencyOut))
+  const renderTokenInfo =
```
Comment (hungdoansy) [inline src/pages/SwapV2/index.tsx:565]: This should be renamed to shouldRenderTokenInfo as it's a boolean

File: src/pages/SwapV2/index.tsx (hunk 3 above)
Comment (nguyenhoaidanh) [reply src/pages/SwapV2/index.tsx:530]: used in `findTokenPairFromUrl` function

**Review by hungdoansy — CHANGES_REQUESTED** [review]
Please take a look at the comments

File: src/pages/SwapV2/index.tsx (hunk 4 above)
Comment (nguyenhoaidanh) [reply src/pages/SwapV2/index.tsx]: okay, updated.

File: src/pages/SwapV2/index.tsx (hunk 3 above)
Comment (hungdoansy) [reply src/pages/SwapV2/index.tsx:530]: Ok, I found it not reasonable, but we can skip for now. We'll do refactoring on this later. Any comment @viet-nv ?

**Review by hungdoansy — APPROVED** [review]
LGTM 👍

File: src/pages/SwapV2/index.tsx (hunk 3 above)
Comment (viet-nv) [reply src/pages/SwapV2/index.tsx:530]: if defaultTokens  is used in `findTokenPairFromUrl` => findTokenPairFromUrl should be wrapped with useCallback, and defaultTokens is a dependency of that useCallback

Comment (viet-nv) [reply src/pages/SwapV2/index.tsx:530]: agree that you can skip it for now but please take a note about this

Comment (nguyenhoaidanh) [reply src/pages/SwapV2/index.tsx:530]: image
I only want to check the length of defaultTokens and return if length is zero. it's okay. I updated it: bring condition to here. is it okay?
image

**Pull request merged**